test/TimeLock.js

```diff
@@ -0,0 +1,180 @@
+    const eta = 12 * 60 * 60;
+
+    await expect(
+      timeLock.scheduleSetNum(HashZero, eta),
+    ).to.be.revertedWithCustomError(timeLock, 'InvalidTimeLockHash');
+  });
+
+  it('should successfully schedule a timelock', async () => {
+    const num = 5;
+    const numParam = defaultAbiCoder.encode(['uint256'], [num]);
+    const numHash = keccak256(numParam);
+    const timeDelay = 13 * 60 * 60;
+
+    const block = await ethers.provider.getBlock('latest');
+    const eta = block.timestamp + timeDelay;
+
+    await timeLock.scheduleSetNum(numHash, eta).then((tx) => tx.wait());
+
+    const etaTimeLock = await timeLock.getTimeLock(numHash);
+    expect(etaTimeLock).to.equal(eta);
+  });
+
+  it('should set the timelock to the minimum time delay if the time delay provided is too low', async () => {
+    const num = 5;
+    const numParam = defaultAbiCoder.encode(['uint256'], [num]);
+    const numHash = keccak256(numParam);
+    const minTimeDelay = 12 * 60 * 60;
+    const timeDelay = 2 * 60 * 60;
+
+    const block = await ethers.provider.getBlock('latest');
+    const eta = block.timestamp + timeDelay;
+
+    await timeLock.scheduleSetNum(numHash, eta).then((tx) => tx.wait());
+
+    const minEta = block.timestamp + minTimeDelay;
+
+    const etaTimeLock = await timeLock.getTimeLock(numHash);
+
+    // allow for small block timestamp disparity
+    expect(Math.abs(etaTimeLock - minEta)).to.be.lessThanOrEqual(5);
+  });
+
+  it('should revert when scheduling the same timelock twice', async () => {
+    const num = 5;
+    const numParam = defaultAbiCoder.encode(['uint256'], [num]);
+    const numHash = keccak256(numParam);
+    const timeDelay = 13 * 60 * 60;
+
+    const block = await ethers.provider.getBlock('latest');
+    const eta = block.timestamp + timeDelay;
+
+    await timeLock.scheduleSetNum(numHash, eta).then((tx) => tx.wait());
+
+    await expect(
+      timeLock.scheduleSetNum(numHash, eta),
+    ).to.be.revertedWithCustomError(timeLock, 'TimeLockAlreadyScheduled');
+  });
+
+  it('should revert when cancelling a timelock with invalid hash', async () => {
+    await expect(timeLock.cancelSetNum(HashZero)).to.be.revertedWithCustomError(
+      timeLock,
+      'InvalidTimeLockHash',
+    );
+  });
+
+  it('should cancel a timelock', async () => {
+    const num = 5;
+    const numParam = defaultAbiCoder.encode(['uint256'], [num]);
+    const numHash = keccak256(numParam);
+    const timeDelay = 13 * 60 * 60;
+
+    const block = await ethers.provider.getBlock('latest');
+    const eta = block.timestamp + timeDelay;
+
+    await timeLock.scheduleSetNum(numHash, eta).then((tx) => tx.wait());
+
+    const etaTimeLock = await timeLock.getTimeLock(numHash);
+    expect(etaTimeLock).to.equal(eta);
+
+    await timeLock.cancelSetNum(numHash).then((tx) => tx.wait());
+
+    const cancelledEta = await timeLock.getTimeLock(numHash);
+    expect(cancelledEta).to.equal(0);
```

Comment:
also check if the timelock can be scheduled again after cancellation

Reply:
Added test case